src/sqlancer/duckdb/DuckDBToStringVisitor.java

```diff
@@ -82,6 +104,72 @@ private void visit(DuckDBSelect select) {
         }
     }
 
+    public void visit(DuckDBResultMap expr) {
+        // use CASE WHEN to express the constant result of a expression
+        Map<DuckDBColumnReference, List<DuckDBExpression>> dbstate = expr.getDbStates();
+        List<DuckDBExpression> result = expr.getResult();
+        int size = dbstate.values().iterator().next().size();
+        if (size == 0) {
+            sb.append(" NULL ");
+            return;
+        }
+        sb.append(" CASE ");
+        for (int i = 0; i < size; i++) {
+            sb.append("WHEN ");
+            Boolean isFirstCondition = true;
+            for (DuckDBColumnReference columnRef : dbstate.keySet()) {
+                if (!isFirstCondition) {
+                    sb.append(" AND ");
+                }
+                visit(columnRef);
+                if (dbstate.get(columnRef).get(i) instanceof DuckDBNullConstant) {
+                    sb.append(" IS NULL");
+                } else if (dbstate.get(columnRef).get(i) instanceof DuckDBTypeCast) {
+                    DuckDBTypeCast ct = (DuckDBTypeCast) dbstate.get(columnRef).get(i);
+                    if (ct.getExpression() instanceof DuckDBNullConstant) {
+                        sb.append(" IS NULL");
+                    } else {
+                        sb.append(" = ");
+                        visit(dbstate.get(columnRef).get(i));
+                    }
+                } else {
+                    sb.append(" = ");
+                    visit(dbstate.get(columnRef).get(i));
+                }
+                isFirstCondition = false;
+            }
+            sb.append(" THEN ");
+            visit(result.get(i));
+            sb.append(" ");
+        }
+        sb.append("END ");
+    }
+
+    public void visit(DuckDBTypeCast expr) {
+        if (expr.getExpression() instanceof DuckDBNullConstant) {
```

Comment:
It seems that we are re-interpreting the AST here (e.g., if it's a `NULL`, we ignore the cast). In the other parts, I think we treat the AST as a ground truth (i.e., we just directly print its structure). Would it be possible to construct the AST earlier on correctly, by omitting casts if we have `null` values or if the type is `unknown`?